python/GafferUI/CompoundEditor.py

```diff
@@ -53,27 +53,28 @@
 
 class CompoundEditor( GafferUI.Editor ) :
 
-	def __init__( self, scriptNode, children=None, detachedPanels=None, windowState=None, **kw ) :
+	# The CompoundEditor constructor args are considered 'private', used only
```

Comment:
Just (attempting to )recall the conversation John and I had before we left. Iirc, its that in the long-term we'd like to move to using `GraphComponents` to store layout, etc... such that we have a mechanism to store any state editors may have. It would also mean we could move to `exec` over `eval`, and so have more robust failure modes. At this point, it'd just be like a normal Gaffer script, and so were piece-meal assemble the layouts as you mentioned. So I think yeah, the inference was that it'd be better to re-build layouts as we do scripts.

Reply:
The `children` argument has always been kindof defacto private, in that it must be passed a data structure that is nontrivial, undocumented, and has changed over time. The only way I've been aware of anyone constructing a CompoundEditor is by building the one they want manually and then saving it as a layout; there are no public APIs to allow a layout to be specified programmatically, and I've encouraged Tom not to add one for now. Ditching the `children` argument in favour of the even more opaque `_state` argument is just a way of making this decision more explicit.

Longer term we have a bigger question about how we should really be serialising these layouts, particularly if they end up serialised in scripts and folks start wanting even more state to be included. That's where I suspect a formal setup involving GraphComponents and our standard serialisation mechanism could be a win over the current ad hoc approach. It's also the point when I think we'd be happier to expose a public API for building layouts. But in the meantime, we're making things more explicitly opaque and private so we have less baggage to deal with in any future transition.